marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/CallBatcherTest.java

```diff
@@ -0,0 +1,336 @@
+	  private static Map<String, JsonNode> endpointdefs = new HashMap<>();
+	    {
+	        ObjectNode servicedef = objectMapper.createObjectNode();
+	        servicedef.put("endpointDirectory", ENDPOINT_DIRECTORY);
+	        serviceHandle = new JacksonHandle(servicedef);
+	    }
+
+	  
+	  private CallManager.CallableEndpoint installEndpoint(String functionName) {
+		  if(functionName == null || functionName.length() == 0)
+			  throw new MarkLogicInternalException("Invalid input was sent.");
+		  JsonNode endpointdef = endpointdefs.get(functionName);
+	      return callMgr.endpoint(serviceHandle, new JacksonHandle(endpointdef), "sjs");
+	  }
+	  
+	  @BeforeClass
+	  public static void setup() {
+		NODE_FORMATS.put("array", Format.JSON);
+		NODE_FORMATS.put("binaryDocument", Format.BINARY);
+		NODE_FORMATS.put("jsonDocument", Format.JSON);
+		NODE_FORMATS.put("object", Format.JSON);
+		NODE_FORMATS.put("textDocument", Format.TEXT);
+		NODE_FORMATS.put("xmlDocument", Format.XML);
+
+		DatabaseClient adminClient = Common.newServerAdminClient("java-unittest-modules");
+		JSONDocumentManager docMgr = adminClient.newJSONDocumentManager();
+
+		DocumentMetadataHandle docMeta = new DocumentMetadataHandle();
+
+		docMeta.getPermissions().add("rest-reader", DocumentMetadataHandle.Capability.EXECUTE);
+		setupNoParamReturnEndpoint(docMgr, docMeta, "noParamReturn", "double", "1.2");
+		setupEndpointSingleRequired(docMgr, docMeta, "oneParam", "double");
+		setupEndpointMultipleRequired(docMgr, docMeta, "float", "manyParam");
+		
+		adminClient.release();
+	  }
+	  
+	  @Test
+	  public void noneForArgsTest() {
+		  CallManager.CallableEndpoint callableEndpoint = installEndpoint("noParamReturn");
+		  CallManager.NoneCaller noneCaller = callableEndpoint.returningNone();
+		  
+		  CallBatcher<CallManager.CallArgs,CallManager.CallEvent> batcher = noneCaller.batcher().forArgs();
+		  assertNotNull(batcher);
+		  ArrayList<String> assignedParams = new ArrayList<String>();
+		  batcher.onCallSuccess(event-> { 
+			  assignedParams.add(event.getArgs().getAssignedParamNames()[0]);
+		  });
+		  
+		  batcher.getDataMovementManager().startJob(batcher);
+		  
+	      batcher.add(noneCaller.args().param("param1", 1.2));
+
+	      batcher.flushAndWait();
+	      assertTrue(assignedParams.size()!=0);
+	      assertTrue(assignedParams.get(0).contains("param1"));
+	  }
+	  
+	  @Test
+	  public void oneForArgsTest() {
+		  CallManager.CallableEndpoint callableEndpoint = installEndpoint("oneParam");
+		  CallManager.OneCaller<Double> oneCaller = callableEndpoint.returningOne(Double.class);
+		  
+		  CallBatcher<CallManager.CallArgs,CallManager.OneCallEvent<Double>> batcher = oneCaller.batcher().forArgs();
+		  assertNotNull(batcher);
+		  ArrayList<String> assignedParams = new ArrayList<String>();
+		  batcher.onCallSuccess(event-> { 
+			  assignedParams.add(event.getArgs().getAssignedParamNames()[0]);
```

Comment:
OneCallEvent<Double> has a Double getItem() method that provides access to the return value.

Could the success event capture that return value in a local variable and confirm it with an assertion? after flushing and waiting so that the test provides more complete coverage of the intended behavior?

The same comment applies to the manyForArgsTest with the difference that ManyCallEvent<Double> has a getItems() method.